### ISSUE TITLE
bug: useReadContracts using wrong client chain id to perform RPC calls

### ISSUE THREAD

**Issue by clumsier** [wagmi-dev/wagmi#3201]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

2.0.0-beta.0

### Current Behavior

Current implementation of `useReadContracts` uses current chain id to perform RPC calls:

https://github.com/wagmi-dev/wagmi/blob/beta/packages/react/src/hooks/useReadContracts.ts#L67

It ignores `chainId` property provided on `contracts` array, eg:
```
useReadContracts({
  contracts: [
    {
      ...getSomeERC20Contract(mainnetChain), // returns abi and address for ERC20 on mainnet chain
      chainId: mainnetChain?.id,
      functionName: 'balanceOf',
      args: [address || ZERO_ADDRESS],
    },
  ],
});
```

It causes an issue that if we have some contract calls on chain with id X and user is connected to chain Y it tries to read from RPC/transport related to chain Y rather than X.

It then throws error that contract which is deployed on chain X doesn't exist on Y (which is true if address differs)

### Expected Behavior

That call above should work regardless of chain that user is connected to.

I could overcome it currently with separating calls into single `useReadContract` calls but it doesn't seem optimal?

### Steps To Reproduce

Try to perform read call eg. any ERC20 balance via `contracts` property on `useReadContracts` hook while being connected to another chain and see RPC network calls going to wrong provider.

**Comment by tmm** [comment]
Thanks for the feedback should be fixed by https://github.com/wagmi-dev/wagmi/pull/3180/commits/c0e191ee064e80fa6413909f2716e96df9b8fecb. Try out `wagmi@2.0.0-beta.1` and let me know!

**Comment by clumsier** [comment]
> Thanks for the feedback should be fixed by [c0e191e](https://github.com/wagmi-dev/wagmi/commit/c0e191ee064e80fa6413909f2716e96df9b8fecb). Try out `wagmi@2.0.0-beta.1` and let me know!

Thanks, it works now :)
